src/data_provider/src/sysInfoLinux.cpp

```diff
@@ -400,6 +408,134 @@ nlohmann::json SysInfo::getNetworks() const
+    {
+        processInfo = statContent.substr(openParenthesisPos + 1, closeParenthesisPos - openParenthesisPos - 1);
+    }
+
+    return processInfo;
+}
+
+void findInodeMatch(char path[], std::vector<int64_t>& inodes, bool& matchInode, const nlohmann::json& JSONInput)
+{
+    // Read symbolic link files.
+    char fdFileContent[KByte] {0};
+
+    if (readlink(path, fdFileContent, sizeof(fdFileContent)) > 0)
+    {
+        if (std::string(fdFileContent).find("socket") != std::string::npos)
+        {
+            // Parse inode from symbolic link. The expected format is: socket[<inode_number>].
+            const int openBracketPos = std::string(fdFileContent).find("[");
+            const int closeBracketPos = std::string(fdFileContent).find("]");
+            const std::string match = std::string(fdFileContent).substr(openBracketPos + 1, closeBracketPos - openBracketPos - 1);
+            const int64_t matchN = std::stoi(match);
+
+            if (std::any_of(JSONInput.cbegin(), JSONInput.cend(), [&](const auto it)
+        {
+            return it.at("inode") == matchN;
+            }))
+            {
+                matchInode = true;
+                inodes.push_back(matchN);
+            }
+        }
+    }
+}
+
+void parseProcFS(const char path[], nlohmann::json& JSONInput)
+{
+    struct dirent* dirent = nullptr;
+    DirPtr procDir(opendir(path));
+
+    // Open proc directory.
+    if (nullptr != procDir.get())
+    {
+        // Iterate over files in proc directory.
+        while (nullptr != (dirent = readdir(procDir.get())))
+        {
+            if ('.' != dirent->d_name[0])
+            {
+                // Proc files path (includes PID directories).
+                char pidPath[KByte] {0};
+                snprintf(pidPath, sizeof(pidPath), "%s/%s", path, dirent->d_name);
+
+                if (Utils::isNumber(dirent->d_name))
+                {
+                    struct dirent* dirent2 = nullptr;
+                    DirPtr pidDir(opendir(pidPath));
+
+                    // Open PID directory.
+                    if (nullptr == pidDir)
+                    {
+                        // Sometimes the PID folder couldn't be open due to library limitation, but it must continue.
+                        continue;
+                    }
+
+                    // Iterate over files in PID directory.
+                    std::vector<int64_t> inodes;
+                    bool matchInode{false};
+
+                    while (nullptr != (dirent2 = readdir(pidDir.get())))
+                    {
+                        if ('.' != dirent2->d_name[0])
+                        {
+                            // Pid files path (includes fd directory and stat file).
+                            char pidFilesPath[KByte] {0};
+                            snprintf(pidFilesPath, sizeof(pidFilesPath), "%s/%s", pidPath, dirent2->d_name);
+
+                            if (!strcmp(dirent2->d_name, "fd"))
```

Comment:
Can we add a comment saying that this algorithm depends on finding the `fd` folder before the `stat` folder?

Reply:
Moved outside loop.